sections/algorithmic_components.tex

```diff
@@ -795,41 +790,59 @@ \subsection{Retrieve Diffim Template for a Visit}
+The mechanics of image subtraction will depend on the choice of algorithm.  The performance of several variations are being considered \citep{Reiss16}.  The algorithms differ primarily in their treatment of correlated noise and whether PSFs are measured directly or a matching kernel is used.
+
+In the classic method of Alard \& Lupton (A\&L) \citep{Alard98}, the
+science image is modeled as a convolution of the template image by a
+PSF--matching kernel $\kappa(u,v;x,y)$, i.e., $S = \kappa \otimes
+T$.\footnote{Indices $u,v$ indicate that the kernel itself is a
+2--dimensional function, which varies as a function of position $x,y$ in
+the image; during convolution and correlation there is an implicit
+summation over $u,v$.} Then the difference image, upon which new or
+variable sources are detected, is given by $D = S - (\kappa \otimes
+T)$.
+We model the PSF--matching kernel by decomposing it into a set of basis
+functions $\kappa(u,v) = \sum_i a_i \kappa_i(u,v)$, where the coefficients
+are determined via ordinary least-squares estimation.  The basis functions
+$\kappa_i(u,v)$ are a degree of freedom in this problem.
+\footnote{The current implementation of the A\&L matching algorithm is
+summarized in detail by Becker et al. (2013) (\url{http://ls.st/x9f}).}
+
+A spatially-invariant matching kernel $\kappa(u,v)$ is determined separately for image substamps centered on multiple kernel candidates across the image after brighter-fatter corrections have been applied.  The kernel candidates are selected using the {\tt DiaCatalogSourceSelector} to query the appropriate reference catalog for sources to use for PSF matching. This selector allows the user to specify the brightness and color range of the objects, toggle star or galaxy selection, and to include variable objects or not. Sources are vetted for signal-to-noise and masked pixels (in both the template and science image). The matching (spatially-invariant) kernel models $\kappa_j(u,v)$, determined for each kernel candidate $j$ as described above, are examined and filtered by various quality measures.
```

Comment:
I'm a little uncomfortable talking about specific class names in this document.  I think it indicates we are too close to the implementation and I'm afraid it will make it hard to maintain.  I think just generalizing to say "a star selector" or something like that.